code/GridFieldDuplicateRow.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * Duplicate the selected row
+ *
+ * @author Gabriele Brosulo <gabriele@brosulo.net>
+ * @creation-date 21-apr-2017
+ */
+class GridFieldDuplicateRow implements GridField_ColumnProvider, GridField_ActionProvider {
```

Comment:
Does this essentially replicate https://github.com/unisolutions/silverstripe-copybutton/blob/master/code/GridFieldCopyButton.php ?